flow/server/flow_server.go

```diff
@@ -208,15 +203,65 @@ func NewFlowServerUDPConn(addr string, port int) (*FlowServerUDPConn, error) {
 	return &FlowServerUDPConn{conn: conn, maxFlowBufferSize: flowsMax}, err
 }
 
-func (s *FlowServer) storeFlows(flows []*flow.Flow) {
-	if len(flows) > 0 {
-		if s.storage != nil {
-			if err := s.storage.StoreFlows(flows); err != nil {
-				logging.GetLogger().Error(err)
-			} else {
-				logging.GetLogger().Debugf("%d flows stored", len(flows))
+// UpdateFlow update the flow from FlowUpdate
+func UpdateFlow(f *flow.Flow, u *flow.FlowUpdate) *flow.Flow {
+	if lup := u.GetLastUpdateMetric(); lup != nil {
+		f.LastUpdateMetric = lup
+	}
+	if m := u.GetMetric(); m != nil {
+		f.Metric = m
+	}
+	if tcp := u.GetTCPMetric(); tcp != nil {
+		f.TCPMetric = tcp
+	}
+	if ip := u.GetIPMetric(); ip != nil {
+		f.IPMetric = ip
+	}
+	f.Last = u.GetLast()
+	f.FinishType = u.GetFinishType()
+
+	if packets := u.GetRawPacketsCaptured(); packets > 0 {
+		f.LastRawPackets = u.LastRawPackets[:0]
+	}
+	return f
+}
+
+func (s *FlowServer) storeFlows(msgs []*flow.Message) {
+	if len(msgs) < 1 || s.storage == nil {
+		return
+	}
+
+	var flows []*flow.Flow
+	for _, msg := range msgs {
+		if len(msg.Flows) > 0 {
+			flows = append(flows, msg.Flows...)
+		}
+		if len(msg.Updates) > 0 {
+			for _, update := range msg.Updates {
+				fsq := filters.SearchQuery{}
+				tsf := filters.NewTermStringFilter("UUID", update.FlowUUID)
```

Comment:
We can request the datastore for each update, this will decrease the performance. I would add a UpdateFlows function to the storage interface.